### PR TITLE
Switch to Coil for image loading

### PR DESCRIPTION
**Pull request by ZacSweers**
Not total feature parity with Glide but pretty close for my needs!

### CODE REVIEW

File: app/src/main/kotlin/io/sweers/catchup/app/ApplicationModule.kt [hunk 1]
```diff
@@ -139,5 +151,88 @@ abstract class ApplicationModule {
...
+        callFactory(object : Call.Factory {
+          override fun newCall(request: Request): Call {
+            return okHttpClient.get().newCall(request)
+          }
+        })
+
+        // Hardware bitmaps don't work with the saturation effect
+        allowHardware(false)
+        allowRgb565(isLowRamDevice)
+        crossfade(true)
```
Comment (colinrtwhite) [inline app/src/main/kotlin/io/sweers/catchup/app/ApplicationModule.kt]: `crossfade(true)` is an alias for `crossfade(100)` (100 is the default duration), similarly `crossfade(false)` = `crossfade(0)`

File: app/src/main/kotlin/io/sweers/catchup/ui/CoilImagesPlugin.kt [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  private class CoilAsyncDrawableLoader(
+    private val context: Context,
+    private val imageLoader: ImageLoader
+  ) : AsyncDrawableLoader() {
+    private val cache = HashMap<AsyncDrawable, Target>(2)
+
+    override fun load(drawable: AsyncDrawable) {
+      val target = AsyncDrawableTarget(drawable)
+      cache[drawable] = target
+      imageLoader.load(context, drawable.destination) {
```
Comment (colinrtwhite) [inline app/src/main/kotlin/io/sweers/catchup/ui/CoilImagesPlugin.kt]: `load` returns a `RequestDisposable`, which could be used to cancel the Request.

File: app/src/main/kotlin/io/sweers/catchup/app/ApplicationModule.kt (hunk 1 above)
Comment (ZacSweers) [reply app/src/main/kotlin/io/sweers/catchup/app/ApplicationModule.kt]: Good to know! Will clean this up

File: app/src/main/kotlin/io/sweers/catchup/ui/CoilImagesPlugin.kt (hunk 2 above)
Comment (ZacSweers) [reply app/src/main/kotlin/io/sweers/catchup/ui/CoilImagesPlugin.kt]: True, I started down this road but got stuck on something at the time. Will revisit

**Pull request merged**